### PR TITLE
Edit and delete comments

### PR DESCRIPTION
**Pull request by Coderinprogress96**
-Make the edit routes put and get  for comments using method override put and create the button edit in the comments.
-Make the delete route with method override in the button and mongoose .findByIdAndRemove method and add the delete button to the comments.

### CODE REVIEW

File: routes/campgrounds.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-const   express = require("express");
-const   router = express.Router();
-        methodOverride = require("method-override"),
-        Campground = require("../models/campground"),
-        Comment = require("../models/comments");
+const express = require("express");
+const router = express.Router();
+methodOverride = require("method-override"),
```
Comment (br1code) [inline routes/campgrounds.js:3]: `methodOverride ` is not being used in this file

File: routes/comments.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
+const campground = require("../models/campground");
+const { route } = require("./campgrounds");
+
 const   express = require("express"),
-        router = express.Router(),
+        router = express.Router({mergeParams: true}),
```
Comment (br1code) [inline routes/comments.js:5]: What does that do? the `mergeParams: true`

File: views/partials/header.ejs [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <meta name="viewport" content="width=device-width">
     <title>YelpCamp</title>
     <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk" crossorigin="anonymous">
-    <link rel="stylesheet" href="/stylesheets/main.css">
+    <link rel="stylesheet" href="/public/stylesheet/main.css">
```
Comment (br1code) [inline views/partials/header.ejs:8]: I like this! I always create a public folder

File: app.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ app.use(function(req, res, next){
 });
 app.use(authRoutes);
 app.use("/campgrounds",campgroundRoutes);
-app.use(commentRoutes);
+app.use("/campgrounds/:id/comments",commentRoutes);
```
Comment (br1code) [inline app.js:48]: not a super fan of this, you could have problems in the future if you continue adding routes to `commentRoutes`. But ... it's fine

File: routes/comments.js (hunk 2 above)
Comment (Coderinprogress96) [reply routes/comments.js:5]: With this we merge the params of the comments with the campgrounds params and we are able to access the campground id in the comments route.

File: routes/campgrounds.js (hunk 1 above)
Comment (Coderinprogress96) [reply routes/campgrounds.js:3]: Thats true i require it in vain, now im deleting it from that file

**Pull request merged**